datastore/save.go

```diff
@@ -138,6 +138,17 @@ func reflectFieldSave(props *[]Property, p Property, name string, opts saveOpts,
 		}
 	}
 
+	if v.CanAddr() {
+		vi := v.Addr().Interface()
+		if pro, ok := vi.(PropertyLoadSaver); ok {
+			res, err := pro.Save()
+			if err != nil {
+				return fmt.Errorf("datastore: unsupported struct field type: %v", v.Type())
+			}
+			p.Value = res
+		}
+	}
+
```

Comment:
How about not to add here checking for implementation of `PropertyLoadSaver`but instead delete [this part](https://github.com/q-logic/google-cloud-go/blob/310f44047c0699edb436207ff0e19f1613b81eb0/datastore/prop.go#L320) in `plsForSave`?:
```
	if v.IsNil() {
		return nil, nil
	}
```

Reply:
Not delete but to `if v.IsNil() && !v.CanAddr() `?